src/caret_analyze/architecture/architecture.py

```diff
@@ -101,16 +101,65 @@ def is_target_comm(comm: CommunicationStructValue):
         return Util.find_one(is_target_comm, self.communications)
 
     def get_path(self, path_name: str) -> PathStructValue:
-        return self._path_manager.get_named_path(path_name)
+        if path_name not in self.path_names:
+            raise InvalidArgumentError(f'Failed to get named path. {path_name} not exist.')
+
+        named_path: PathStruct = Util.find_one(lambda x: x._path_name == path_name, self._paths)
+        return named_path.to_value()
 
     def add_path(self, path_name: str, path_info: PathStructValue) -> None:
-        self._path_manager.add_named_path(path_name, path_info)
+        if path_name in self.path_names:
+            raise InvalidArgumentError('Failed to add named path. Duplicate path name.')
+
+        child: List[Union[NodePathStruct, CommunicationStruct]] = []
+
+        for c in path_info.child:
+            if isinstance(c, NodePathStructValue):
+                from .graph_search import NodePathSearcher
+                node_path_searcher: NodePathSearcher = \
+                    NodePathSearcher(tuple(self._nodes), tuple(self._communications))
+
+                node_path: NodePathStruct = \
+                    node_path_searcher._find_node_path('' if c.subscribe_topic_name is None
+                                                       else c.subscribe_topic_name,
+                                                       '' if c.publish_topic_name is None
+                                                       else c.publish_topic_name, c.node_name)
+                child.append(node_path)
+
+            elif isinstance(c, CommunicationStructValue):
+                def is_target_comm(comm: CommunicationStruct):
+                    return comm.publish_node_name == c.publish_node_name and \
+                        comm.subscribe_node_name == c.subscribe_node_name and \
+                        comm.topic_name == c.topic_name
+
+                comm: CommunicationStruct = \
+                    Util.find_one(is_target_comm, self._communications)
+                child.append(comm)
+
+            else:
+                raise UnsupportedTypeError('')
+
+        named_path_info = PathStruct(path_name, child)
+        self._paths.append(named_path_info)
```

Comment:
The process that received PathStructValue convert and append to self._paths (List[PathStruct]) is implemented.